qiskit/transpiler/passes/layout/disjoint_utils.py

```diff
@@ -111,7 +111,10 @@ def split_barriers(dag: DAGCircuit):
             barrier_uuid = f"{node.op.label}_uuid={uuid.uuid4()}"
         else:
             barrier_uuid = f"_none_uuid={uuid.uuid4()}"
-        split_dag = DAGCircuit()
+        split_dag = DAGCircuit(
+            node_count_hint=3 * num_qubits,
+            edge_count_hint=2 * num_qubits,
+        )
```

Comment:
Is this just a guess, or is there a reason to it?  Maybe not worth bothering if it's a guess.

Reply:
See my answer to Elena's earlier question about this: https://github.com/Qiskit/qiskit/pull/12812#discussion_r1691320571

Reply:
Whoops, sorry.